### PR TITLE
tracy: add capture and update binaries

### PR DESCRIPTION
**Pull request by nagisa**
#### Motivation for this change

These are used to capture the trace without using the GUI tool and
update the saved trace files from older to newer versions. `update` also
allows to re-compress the trace files for whatever reason one might want
to do it.

Based on top of #91228 

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

cc @mpickering @cpcloud

### CODE REVIEW

**Review by cpcloud — APPROVED** [review]
LGTM

File: pkgs/development/tools/tracy/default.nix [hunk 1]
```diff
@@ -1,34 +1,38 @@
-{ stdenv, lib, darwin, fetchFromGitHub, tbb, gtk2, glfw, pkgconfig, freetype, Carbon, AppKit }:
+{ stdenv, lib, darwin, fetchFromGitHub, tbb, gtk2, glfw, pkgconfig, freetype, Carbon, AppKit, capstone }:
 
 stdenv.mkDerivation rec {
-  name    = "tracy-${version}";
-  version = "0.6.3";
+  name = "tracy-${version}";
```
Comment (danieldk) [inline pkgs/development/tools/tracy/default.nix:4]: Maybe switch to `pname`

```suggestion
  pname = "tracy";
```

while at it?

Comment (nagisa) [reply pkgs/development/tools/tracy/default.nix:4]: :+1:

**Comment by danieldk** [comment]
Maintainer ping: @mpickering

**Comment by mpickering** [comment]
Looks fine to me. I didn't add them originally as I don't use them.

**Pull request merged**